src/main/kotlin/com/openlattice/mechanic/upgrades/ResetEntitySetCountsMaterializedView.kt

```diff
@@ -0,0 +1,35 @@
+package com.openlattice.mechanic.upgrades
+
+import com.openlattice.data.storage.PostgresEntitySetSizesInitializationTask
+import com.openlattice.mechanic.Toolbox
+import org.slf4j.LoggerFactory
+
+class ResetEntitySetCountsMaterializedView(val toolbox: Toolbox) : Upgrade {
+
+    companion object {
+        private val logger = LoggerFactory.getLogger(ResetEntitySetCountsMaterializedView::class.java)
+    }
+
+    override fun upgrade(): Boolean {
+
+        logger.info("About to drop and recreate view ${PostgresEntitySetSizesInitializationTask.ENTITY_SET_SIZES_VIEW}.")
+
+        toolbox.hds.connection.use { conn ->
+            conn.autoCommit = false
```

Comment:
I've had issues with setting autocommit to false within the .use{} block instead of outside of it. Is that issue not happening here?

Reply:
I think in the case you had where you had to set it outside of `.use{}` it was `dataSource.connection.createStatement().use { stmt ->` which holds the created statement for the duration of `.use{}` whereas this one holds a reference to the connection `toolbox.hds.connection`. So in this case you can just set things on `conn` with no problem but since `toolbox.hds.connection` calls `getConnection`, you're getting a connection from the Hikari pool, so you need to have a reference to the same connection you're calling `.use{}` on. 
I'm sure there's a better way to word all of this but I'm low-caffeine right now 💀

Reply:
Lol makes sense to me! Ty for clarifying!